qa/tests/src/org/jboss/jbossts/qa/CrashRecovery12Resources/Resource01.java

```diff
@@ -87,6 +87,14 @@ public int prepare(Xid xid) throws XAException
 		}
 		if (crashIn == ROLLBACK)
 		{
+			/*
+			 * Since JBTM-3843, rollback is no longer invoked
+			 * during the second phase of the 2PC protocol as
+			 * the Resource Manager should have rolled back
+			 * the branch during the first phase
+			 */
+			myLog(ERR, "Crashing in prepare");
+			myLog(OUT, "Passed");
```

Comment:
I understand the need to change this compared to the existing code, that being said I think the change to this test could be a little more than the lines here because (I presume) this test is testing a crash situation in the resource where the resource has rolled back in prepare but actually the Resource is no longer crashing, rather it is returning XAException.XA_RBROLLBACK out of prepare (as it was doing so before) so I am not sure what to suggest. I presume we maybe should be System.exit(0) in here but not sure what other changes might then be needed.

Reply:
Modified. Thanks Tom